core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -1166,6 +1166,8 @@ final class ShuffleBlockFetcherIterator(
       case ShuffleBlockBatchId(shuffleId, mapId, startReduceId, _) =>
         throw SparkCoreErrors.fetchFailedError(address, shuffleId, mapId, mapIndex, startReduceId,
           msg, e)
+      case ShuffleBlockChunkId(_, _, _, _) =>
+        pushBasedFetchHelper.initiateFallbackFetchForPushMergedBlock(blockId, address)
```

Comment:
Wouldn't result in duplicate records? Some of the chunk is already consumed but we don't know how much is that correct? So now we fallback to original blocks but and that could re-fetch the records that were already consumed.

Reply:
Can you change it to throwing the exception in this case @zhouyejoe ?

Reply:
Sure, will update accordingly